### ISSUE TITLE
Support custom configuration for specific namespaces

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1398]
**Is your feature request related to a problem? Please describe.**
Nubank has a common kondo config for all services and libs in a single lib which exports the kondo configuration.
We also have a common `user` namespace  which has some requires to make easy the repl usage as follow:
`user.clj`
```clojure
(ns user
  (:require [clojure.tools.namespace.repl :refer [refresh refresh-all]]
            [common-something.misc :as misc]))
```
The issue is that kondo report those namespaces as unused, we want to suppress those linters just for this namespace and via config, as we have hundreds of services changing one single place instead of all services would be the best for us.

**Describe the solution you'd like**
As discussed [here](https://clojurians.slack.com/archives/CHY97NXE2/p1633534053285600) @borkdude suggested having a way to configure kondo configuration for specific namespaces like:

`.clj-kondo/config.edn`
```clojure
{:ns-config {user {:unused-namespaces {...}}}}
```

**Describe alternatives you've considered**
As the last resource, add a `:exclude-ns` to some linters?

**Additional context**
Add support for regex on the namespace would be awesome as we have some kind of pattern for some namespaces like `<service-name>.config`.

**Comment by borkdude** [comment]
I believe there's overlap between this issue and https://github.com/clj-kondo/clj-kondo/issues/1618

**Comment by ericdallo** [comment]
Yes, taking a quick look it seems that one would solve this one too, feel free to close if so
